api/src/main/java/org/infinispan/api/InfinispanManager.java

```diff
@@ -0,0 +1,29 @@
+package org.infinispan.api;
+
+import org.infinispan.api.configuration.InfinispanClientConfig;
+import org.infinispan.api.configuration.InfinispanConfig;
+
+/**
+ * Entry point for Infinispan API - client
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @since 10.0
+ */
+@Experimental
+public class InfinispanManager {
+
+   public static final Infinispan newInfinispanClient(InfinispanClientConfig config) {
+      try {
+         return (Infinispan) InfinispanManager.class.getClassLoader()
+               .loadClass("org.infinispan.api.client.impl.InfinispanClientImpl")
+               .getConstructor(InfinispanClientConfig.class)
+               .newInstance(config);
+      } catch (Exception e) {
+      }
+      return null;
+   }
+
+   public static final Infinispan newInfinispan(InfinispanConfig config) {
```

Comment:
If we're going down the route of having a separate entry point for client and embedded, then IMO I think this name should be more explicit then just `infinispan` as that's very vague.

Reply:
I'm removing the class